accounting/accounting.go

```diff
@@ -94,11 +94,19 @@ func bytesAreZero(b []byte) bool {
 	return true
 }
 
-func (accounting *AccountingState) updateAlgo(addr types.Address, d int64) {
+func (accounting *AccountingState) updateAlgo(addr types.Address, amount int64) {
+	accounting.updateAlgoAndRewards(addr, amount, 0)
+}
+
+func (accounting *AccountingState) updateAlgoAndRewards(addr types.Address, amount, rewards int64) {
 	if accounting.AlgoUpdates == nil {
-		accounting.AlgoUpdates = make(map[[32]byte]int64)
+		accounting.AlgoUpdates = make(map[[32]byte]*idb.AlgoUpdate)
+	}
+	if _, hasKey := accounting.AlgoUpdates[addr]; !hasKey {
```

Comment:
If we're going to do the lookup, we should use the value if it exists instead of doing 3 lookups and maybe one insert. Since the map value is a pointer type this could be one lookup and maybe one insert.

Reply:
Something like this?
```
	var acctUpdate *idb.AlgoUpdate
	var hasKey bool
	if acctUpdate, hasKey = accounting.AlgoUpdates[addr]; !hasKey {
		acctUpdate = &idb.AlgoUpdate{}
		accounting.AlgoUpdates[addr] = acctUpdate
	}
	acctUpdate.Balance = accounting.AlgoUpdates[addr].Balance + amount
	acctUpdate.Rewards = accounting.AlgoUpdates[addr].Rewards + rewards
```

Reply:
yeah but you can just assign to the map with an AlgoUpdate{amount,rewards} and then return.
and `acctUpdate.Balance += amount`